lib/sentry/transport/rate_limiter.ex

```diff
@@ -0,0 +1,174 @@
+defmodule Sentry.Transport.RateLimiter do
+  @moduledoc false
+  # Tracks rate limits per category from Sentry API responses.
+  # Uses an ETS table to store expiry timestamps for rate-limited categories.
+  # When Sentry returns a 429 response with rate limit headers, this module
+  # stores the expiry time per category, allowing other parts of the SDK to
+  # check if an event should be dropped before sending.
+  #
+  # See https://develop.sentry.dev/sdk/expected-features/rate-limiting/
+
+  use GenServer
+
+  @table __MODULE__
+  @sweep_interval_ms 60_000
+
+  ## Public API
+
+  @doc """
+  Starts the RateLimiter GenServer.
+  """
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(opts \\ []) do
+    GenServer.start_link(__MODULE__, nil, name: Keyword.get(opts, :name, __MODULE__))
+  end
+
+  ## GenServer Callbacks
+
+  @impl true
+  def init(nil) do
+    _table = :ets.new(@table, [:named_table, :public, :set])
+    schedule_sweep()
+    {:ok, :no_state}
+  end
+
+  @impl true
+  def handle_info(:sweep, state) do
+    now = System.system_time(:second)
+
+    # Match spec: select entries where expiry (position 2) < now
+    match_spec = [{{:"$1", :"$2"}, [{:<, :"$2", now}], [true]}]
+
+    :ets.select_delete(@table, match_spec)
+
+    schedule_sweep()
+    {:noreply, state}
+  end
+
+  ## Public Functions
+
+  @doc """
+  Checks if the given category is currently rate-limited.
+
+  Returns `true` if the category is rate-limited (either specifically or via
+  a global rate limit), `false` otherwise.
+
+  ## Examples
+
+      iex> RateLimiter.rate_limited?("error")
+      false
+
+      iex> :ets.insert(RateLimiter, {"error", System.system_time(:second) + 60})
+      iex> RateLimiter.rate_limited?("error")
+      true
+
+  """
+  @spec rate_limited?(String.t()) :: boolean()
+  def rate_limited?(category) do
```

Comment:
Categories here are a fixed set, right? Maybe we could be more specific on the strings we expect here?